pkg/validation/options.go

```diff
@@ -271,14 +271,13 @@ func parseProviderInfo(o *options.Options, msgs []string) []string {
 			msgs = append(msgs, "oidc provider requires an oidc issuer URL")
 		}
 	case *providers.GitLabProvider:
-		p.Groups = o.Providers[0].GitLabConfig.Group
-		err := p.AddProjects(o.Providers[0].GitLabConfig.Projects)
+		p.SetAllowedGroups(o.Providers[0].GitLabConfig.Group)
+		err := p.SetAllowedProjects(o.Providers[0].GitLabConfig.Projects)
 		if err != nil {
 			msgs = append(msgs, "failed to setup gitlab project access level")
 		}
-		p.SetAllowedGroups(p.PrefixAllowedGroups())
-		p.SetProjectScope()
 
+		// TODO (@NickMeves) - This needed anymore with GitLab embedding OIDC?
```

Comment:
Is this codepath still needed for GitLab users who don't setup OIDC Issuer details to their specific instance? Defaults to cloud hosted if not set?

 If we still need to account for defaulting to cloud-hosted GitLab if OIDC flags aren't set: I wonder if these values should be moved to the Provider itself like other Providers do?

Reply:
I'll suggest to go with something looking like [this](https://github.com/oauth2-proxy/oauth2-proxy/blob/gitlab-oidc/providers/azure.go#L89). Best UX is, imo, using [gitlab.com](https://gitlab.com) as default and override if needed :

```go
// Configure defaults the AzureProvider configuration options
func (p *AzureProvider) Configure(tenant string) {
	if tenant == "" || tenant == "common" {
		// tenant is empty or default, remain on the default "common" tenant
		return
	}

	// Specific tennant specified, override the Login and RedeemURLs
	p.Tenant = tenant
	overrideTenantURL(p.LoginURL, azureDefaultLoginURL, tenant, "authorize")
	overrideTenantURL(p.RedeemURL, azureDefaultRedeemURL, tenant, "token")
}
```

Reply:
I'm gonna punt on this for now. I think this needs a much broader cleanup/refactor of how OIDC discovery works in configuration and how various providers utilize it.